driver-core/src/main/com/mongodb/internal/operation/AggregateOperationImpl.java

```diff
@@ -237,10 +239,11 @@ BsonDocument getCommand(final SessionContext sessionContext, final int maxWireVe
     }
 
     private QueryResult<T> createQueryResult(final BsonDocument result, final ConnectionDescription description) {
+        assertNotNull(result);
         return cursorDocumentToQueryResult(result.getDocument(CURSOR), description.getServerAddress());
     }
 
-    private CommandReadTransformer<BsonDocument, AggregateResponseBatchCursor<T>> transformer() {
+    private CommandReadTransformer<BsonDocument, QueryBatchCursor<T>> transformer() {
```

Comment:
Used the implementation rather than the interface name (as per the norm across the code).